File: android/src/main/java/app/capgo/plugins/env/EnvPlugin.java
package app.capgo.plugins.env;

import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

@CapacitorPlugin(name = "Env")
public class EnvPlugin extends Plugin {

    private final String pluginVersion = "8.1.6";

    @PluginMethod
    public void getKey(PluginCall call) {
        String key = call.getString("key");

        JSObject ret = new JSObject();
        ret.put("value", this.getConfig().getString(key, ""));
        call.resolve(ret);
    }

    @PluginMethod
    public void getPluginVersion(final PluginCall call) {
        try {
            final JSObject ret = new JSObject();
            ret.put("version", this.pluginVersion);
            call.resolve(ret);
        } catch (final Exception e) {
            call.reject("Could not get plugin version", e);
        }
    }
}

File: ios/Sources/EnvPlugin/EnvPlugin.swift
import Foundation
import Capacitor

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(EnvPlugin)
public class EnvPlugin: CAPPlugin, CAPBridgedPlugin {
    private let pluginVersion: String = "8.1.0"
    public let identifier = "EnvPlugin"
    public let jsName = "Env"
    public let pluginMethods: [CAPPluginMethod] = [
        CAPPluginMethod(name: "getKey", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "getPluginVersion", returnType: CAPPluginReturnPromise)
    ]

    @objc func getKey(_ call: CAPPluginCall) {
        let key = call.getString("key") ?? ""
        call.resolve([
            "value": getConfig().getString(key, "")
        ])
    }

    @objc func getPluginVersion(_ call: CAPPluginCall) {
        call.resolve(["version": self.pluginVersion])
    }

}
